### ISSUE TITLE
vertical alignment for \rule seems off

### ISSUE THREAD

**Issue by tscholl2** [Khan/KaTeX#646]
When a `\rule` has a height argument, its vertical alignment seems to be off:

![image](https://cloud.githubusercontent.com/assets/8905985/22275099/2ac2d8ce-e25f-11e6-9ac6-8812a7e708b5.png)

You can see the issue without `\frac` or `\color`, I just used them to highlight the issue.

Here is a similar expression in latex:

![image](https://cloud.githubusercontent.com/assets/8905985/22275135/6be33f92-e25f-11e6-92c2-4193bf587e0d.png)

rendered with:
```
\documentclass{article}
\usepackage{color}
\begin{document}
$$
\frac{\textcolor{blue}{\rule{2em}{2em}}}{2}
$$
\end{document}
```

The katex screenshot was taken from the katex homepage today, and I've seen the same thing running version `0.7.1` (I'm not sure if the github pages has the most up to date version).

I'd love to help fix this if possible, but my fear is that it's a very subtle and complicated CSS issue. My guess is to look at the rule for `\rule` in the `buildHTML`? Specifically, I thought this function might be important:

https://github.com/Khan/KaTeX/blob/master/src/buildHTML.js#L1319

I don't know if this is a known issue, I couldn't find it searching, so sorry if this is a duplicate.

**Comment by arnog** [comment]
I think the fix might be to set the `style.marginTop` property to `-(height - shift)` and to not set `style.bottom`.

**Comment by ronkok** [comment]
I believe that this issue was resolved by PR #768.
![rule](https://user-images.githubusercontent.com/16403058/29300790-90c699e6-812c-11e7-99d3-2844c7b03e03.PNG)
